legion/engine/physics/physx/physx_event_process_funcs.cpp

```diff
@@ -38,14 +38,12 @@ namespace legion::physics
 
             convexCollider.setRegistered(true);
 
-            PxTransform transform;
             PxTransform localTransform;
-
-            calculateGlobalAndLocalTransforms(localTransform, transform, convexCollider, entity);
+            calculateLocalColliderTransform(localTransform, convexCollider);
 
             const math::vec3& extents = convexCollider.getBoxExtents();
 
-            instantiateNextCollider<PxBoxGeometry,const PxVec3&>(getSDK(), wrapper, transform, localTransform, sceneInfo, PxVec3(extents.x, extents.y, extents.z));
+            instantiateNextCollider<PxBoxGeometry,const PxVec3&>(getSDK(), wrapper, localTransform, sceneInfo, PxVec3(extents.x, extents.y, extents.z));
```

Comment:
ngl these templated calls that require `XXX&` or whatever seem kindah strange to me? can these not be automatically deduced using template parameter deduction? this feels kindah bug prone?

Reply:
Unfortunately this happens when when you dont include the extra parameter:
`1>lld-link : error : undefined symbol: void __cdecl legion::physics::instantiateDynamicActorWith<class physx::PxBoxGeometry, class physx::PxVec3>(class physx::PxPhysics *, struct legion::physics::PhysxInternalWrapper &, class physx::PxTransform const &, class physx::PxTransform const &, struct legion::physics::PhysxEnviromentInfo const &, struct legion::core::ecs::entity, class physx::PxVec3 &&)
1>>>> referenced by D:\Engines\Legion-Engine\legion\engine\physics\physx\physx_event_process_funcs.cpp:69`
From what I understand ig the param was deduced as  `class physx::PxVec3` when it should be `'const class physx::PxVec3&` ?